### PR TITLE
HDDS-8223. SCM delete block service shoud run wait for safemode to exit.

### PR DESCRIPTION
**Pull request by guohao-rosicky**
## What changes were proposed in this pull request?

SCM delete block service shoud run wait for safemode to exit.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-8223

### CODE REVIEW

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/SCMBlockDeletingService.java [hunk 1]
```diff
@@ -225,7 +235,16 @@ public void notifyStatusChanged() {
   public boolean shouldRun() {
     serviceLock.lock();
     try {
-      return serviceStatus == ServiceStatus.RUNNING;
+      long alreadyWaitTimeInMillis = clock.millis() - lastTimeToBeReadyInMillis;
```
Comment (sodonnel) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/SCMBlockDeletingService.java:238]: Could we rename:

lastTimeToBeReadyInMillis -> safemodeExitMillis
waitTimeInMillis -> safemodeExitRunDelayMillis

**Review by sodonnel — COMMENTED** [review]
Looks mostly good. I just suggested a few renames which I think makes the code more readable. Please have a look and let me know what you think.

**Comment by guohao-rosicky** [comment]
> Looks mostly good. I just suggested a few renames which I think makes the code more readable. Please have a look and let me know what you think.

Your suggestion is very good. I have revised it

**Review by sodonnel — APPROVED** [review]
LGTM. I will commit if the check jobs come back green.

**Comment by adoroszlai** [comment]
Seems like checkstyle violations need to be fixed.

```
hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/BlockManagerImpl.java
 21: Unused import - java.time.Duration.
 27: Unused import - java.util.concurrent.TimeUnit.
 30: Unused import - org.apache.hadoop.hdds.HddsConfigKeys.
 37: Unused import - org.apache.hadoop.hdds.scm.ScmConfig.
 54: Unused import - org.apache.hadoop.ozone.OzoneConfigKeys.OZONE_BLOCK_DELETING_SERVICE_TIMEOUT.
 55: Unused import - org.apache.hadoop.ozone.OzoneConfigKeys.OZONE_BLOCK_DELETING_SERVICE_TIMEOUT_DEFAULT.
hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/SCMBlockDeletingService.java
 21: Unused import - java.time.Duration.
 95: More than 7 parameters (found 8).
```

**Comment by sodonnel** [comment]
Note that you can run checkstyle on your laptop easily by running the following from the root of the repo:

```
./hadoop-ozone/dev-support/checks/checkstyle.sh 
```

**Review by sodonnel — APPROVED** [review]

**Pull request merged**